pallets/crowdloans/src/lib.rs

```diff
@@ -153,44 +153,38 @@ pub mod pallet {
     #[pallet::generate_deposit(pub(super) fn deposit_event)]
     pub enum Event<T: Config> {
         /// New vault was created
-        /// [para_id, ctoken_id]
-        VaultCreated(ParaId, AssetIdOf<T>),
+        /// [para_id, vault_id, ctoken_id, phase, contribution_strategy, cap, end_block, trie_index]
+        VaultCreated(
+            ParaId,
+            u32,
+            AssetIdOf<T>,
+            VaultPhase,
+            ContributionStrategy,
+            BalanceOf<T>,
+            BlockNumberFor<T>,
+            u32,
+        ),
         /// Existing vault was updated
         /// [para_id, vault_id, cap, end_block, contribution_strategy]
         VaultUpdated(
             ParaId,
             u32,
+            ContributionStrategy,
             BalanceOf<T>,
             BlockNumberFor<T>,
-            ContributionStrategy,
         ),
         /// Vault was opened
-        /// [para_id]
-        VaultOpened(ParaId),
-        /// Vault was closed
-        /// [para_id]
-        VaultClosed(ParaId),
-        /// Vault was reopened
-        /// [para_id]
-        VaultReOpened(ParaId),
-        /// Vault is successful
-        /// [para_id]
-        VaultSucceeded(ParaId),
-        /// Vault is failing
-        /// [para_id]
-        VaultFailed(ParaId),
-        /// Vault is expiring
-        /// [para_id]
-        VaultExpired(ParaId),
+        /// [para_id, vault_id, pre_phase, now_phase]
+        VaultPhaseChanged(ParaId, u32, VaultPhase, VaultPhase),
         /// Vault is trying to do contributing
         /// [para_id, contributor, amount, referral_code]
         VaultDoContributing(ParaId, T::AccountId, BalanceOf<T>, Vec<u8>),
         /// Vault is trying to do withdrawing
         /// [para_id, amount, target_phase]
         VaultDoWithdrawing(ParaId, BalanceOf<T>, VaultPhase),
```

Comment:
we can also add vault id to make things more clear although it's not useful for XCM